File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-3.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if(len(nums)==0):
            return
        insert_index=1

        for i in range(1, len(nums),1):
            if(nums[i] != nums[i-1]):
                nums[insert_index]=nums[i]
                insert_index += 1
            
        return insert_index

File: Data Structures & Algorithms/search-insert-position/submission-3.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int end = nums.size()-1;
        int st = 0;
        while(st<=end){
            int mid = st + (end-st)/2;
            if(nums[mid]==target){
                return mid;
            }
            else if (nums[mid]>target){
                end = mid -1;
            }
            else{
                st = mid + 1;
            }
        } 
        nums.push_back(target);
        sort(nums.begin(),nums.end());
        for(int i = 0 ; i < nums.size(); i++){
            if(target == nums[i]){
                return i;
            }
        }
        return 0;
    }
};